### PR TITLE
feat(IoTDA): add a new resource to manage resource space

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
add a new resource to manage IoTDA resource space

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/iotda' TESTARGS='-run=TestAccSpace_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run=TestAccSpace_basic -timeout 360m -parallel 4
=== RUN   TestAccSpace_basic
--- PASS: TestAccSpace_basic (23.94s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     24.045s
```

### CODE REVIEW

File: docs/resources/iotda_space.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Changing this parameter will create a new resource.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID in UUID format.
+
+* `is_default` - Whether it is the default resource space. The IoT platform automatically creates and assigns
+a default resource space (undeletable) to your account.
```
Comment (Jason-Zhang9309) [inline docs/resources/iotda_space.md:39]: does that means the space created by user will never be default?

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_space_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testSpace_basic(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "name", name),
```
Comment (niuzhenguo) [inline huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_space_test.go:45]: Nits: suggest to add `is_default` check here as well

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_space_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+				),
+			},
+			{
+				Config: testSpace_basic(updateName),
```
Comment (niuzhenguo) [inline huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_space_test.go]: name is not updatable as described in docs

File: docs/resources/iotda_space.md (hunk 1 above)
Comment (niuzhenguo) [reply docs/resources/iotda_space.md:39]: seems yes, we also have such attribute in other resources. it's ok with me.

Comment (hmbig2) [reply docs/resources/iotda_space.md:39]: > does that means the space created by user will never be default?

yes

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_space_test.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_space_test.go]: ok，removed

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_space_test.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_space_test.go:45]: added

**Pull request merged**